core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BCFile.java

```diff
@@ -672,11 +674,23 @@ public BlockRegion getBlockRegion() {
       }
 
       public void finish() throws IOException {
-        try {
-          in.close();
-        } finally {
-          compressAlgo.returnDecompressor(decompressor);
-          decompressor = null;
+        if (!closed) {
```

Comment:
In general synchronizing on a monitor is significantly slower than checking a boolean.  However since the close method is invoked in general only once on each instance, I can see the benefit of avoiding the initial check.  As Josh pointed out, closes are much less frequent and hence the performance impact is minimal.  I will take it out however.